pkg/build/builder/cmd/scmauth/sshkey.go

```diff
@@ -13,25 +14,25 @@ type SSHPrivateKey struct{}
 
 // Setup creates a wrapper script for SSH command to be able to use the provided
 // SSH key while accessing private repository.
-func (_ SSHPrivateKey) Setup(baseDir string) error {
+func (_ SSHPrivateKey) Setup(baseDir string) (*url.URL, error) {
 	script, err := ioutil.TempFile("", "gitssh")
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer script.Close()
 	if err := script.Chmod(0711); err != nil {
-		return err
+		return nil, err
 	}
 	if _, err := script.WriteString("#!/bin/sh\nssh -i " +
 		filepath.Join(baseDir, SSHPrivateKeyMethodName) +
 		" -o StrictHostKeyChecking=false \"$@\"\n"); err != nil {
-		return err
+		return nil, err
 	}
 	// set environment variable to tell git to use the SSH wrapper
 	if err := os.Setenv("GIT_SSH", script.Name()); err != nil {
-		return err
+		return nil, err
 	}
-	return nil
+	return nil, nil
```

Comment:
From the git doc, it doesn't look like you can... git will only pick up the ssh private key if using a url like user@host:/path which shouldn't contain a password